Yatima/Common/GenTypechecker.lean

```diff
@@ -11,10 +11,15 @@ def genTypechecker : IO $ Except String Expr := do
   Lean.setLibsPaths
   return Yatima.CodeGen.codeGen (← Lean.runFrontend tcCode default) `tc
 
--- TODO: Getting the
+/-- (= (tc decl) 1) ; tc being an expression -/
+def mkRawTypecheckingExpr (tc : Expr) (decl : Digest) : Expr :=
+  Expr.op₂ .numEq
+    (.app tc (.atom $ .commit decl))
+    (.atom $ .num $ .ofNat 1)
 
-def mkRawTypecheckingExpr (tc : Expr) (decl : Digest) : Expr := sorry
-  -- ⟦(= $(Expr.app tc ⟦#c$decl⟧) 1)⟧
-
-def mkCommTypecheckingExpr (tc decl : Digest) : Expr := sorry
-  -- ⟦(= ((eval (open $tc)) $decl) 1)⟧
+/-- (= (tc decl) 1) ; tc being a commitment -/
+def mkCommTypecheckingExpr (tc decl : Digest) : Expr :=
+  Expr.op₂ .numEq
+    (.app (.atom $ .commit tc)
+          (.atom $ .commit decl))
+    (.atom $ .num $ .ofNat 1)
```

Comment:
Using commitments as callable objects is supported since Lurk Beta:
```lisp
lurk-user> (commit (lambda (x) (+ x 1)))
[2 iterations] => #c0x1f4971f046cd9b037eadbc3fcde177431a712e8cef82fcec9e010fc5a65891
lurk-user> (#c0x1f4971f046cd9b037eadbc3fcde177431a712e8cef82fcec9e010fc5a65891 9)
[6 iterations] => 10
```